ansible-ipi-install/roles/node-prep/tasks/10_validation.yml

```diff
@@ -118,6 +118,18 @@
     msg: "The Wildcard (Ingress) VIP is {{ ingressvip }}"
     verbosity: 2
   tags: validation
+#
+- name: Test the connectivity for VIP's
+  command: ping -c3 {{ apivip }}
+  register: results
+  tags: validation
+
+- name: If VIP's are being used fail with an error message
+  fail:
+    msg: "{{ apivip }} and {{ ingressvip }} are being used in somewhere else, kindly turn off the nodes to re-run the deployment"
+  when: '"0% packet loss" in results.stdout'
```

Comment:
I would check the return code just and add a Wait to the ping like:

ping -c3 -W 2 {{ apivip }} so that it waits '2' for each one of those 3 comm, and use the rerturn code, if it's '1' it failed, so the VIP is down, if it's '0' fail because the ip is up.

Also, does this work for both IPv4 and IPv6?

Reply:
Couple things 
* We probably want to have this check happen just before we do the 60_* script. Reasoning is if you did a deployment and where about to deploy again, this would fail 100% of the time because the power off task has not happened yet
* The when statement checks whether you have 0% packet loss, but since you are capturing ping 3 times its possible (though not likely) that you could have a 0% packet loss in 1st message and then all of a sudden connectivity in the other 2 messages. In other words, we need to find  a better when statement to check against IMO
* Unsure if this would work on IPv4 and IPv6 as mentioned by @iranzo

Reply:
Jut confirmed this works for ipv6 as well, will look for more use cases as well. 

```
fatal: [virthost.ostest.test.metalkube.org]: FAILED! => {"changed": false, "msg": "fd35:919d:4042::1121 and 192.168.111.4 are being used in somewhere else, kindly off the nodes to re-run the deployment"}
```

Reply:
10_validation changes should be reverted.